src/components/settings/SettingsSSOTab/hooks.js

```diff
@@ -102,25 +116,63 @@ const useIdpState = () => {
     metadataURL,
     entryType,
     entityID,
+    ssoUrl,
+    publicKey,
     handleMetadataURLUpdate,
     handleMetadataEntryTypeUpdate,
     handleEntityIDUpdate,
     createOrUpdateIdpRecord,
   };
 };
 
+const useExistingProviderData = (enterpriseUuid, refreshBool) => {
```

Comment:
hook to easily fetch and set the customer's existing provider data config, similar to the provider config